applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+    {
+        mInitializeWasPerformed = val;
+    }
+
+    bool GetIsInitialized() const
+    {
+        return mInitializeWasPerformed;
+    }
+
+    SchemePointerType &pGetScheme()
+    {
+        return mpScheme;
+    };
+
+    BuilderAndSolverPointerType &pGetEigenSolver()
+    {
+        return mpEigenSolver;
+    };
+
+    BuilderAndSolverPointerType &pGetBuilderAndSolver()
+    {
+        return mpBuilderAndSolver;
+    };
+
+    // SparseMatrixPointerType &pGetMassMatrix()
+    // {
+    //     return mpMassMatrix;
+    // }
+
+    SparseMatrixPointerType &pGetStiffnessMatrixPrevious()
+    {
+        return mpStiffnessMatrixPrevious;
+    }
+
+    SparseMatrixPointerType &pGetStiffnessMatrix()
+    {
+        return mpStiffnessMatrix;
+    }
+
+    SparseVectorPointerType &pGetRHS()
+    {
+        return mpRHS;
+    }
+
+    SparseVectorPointerType &pGetDx()
+    {
+        return mpDx;
+    }
+
+    // SparseMatrixType &GetMassMatrix()
+    // {
+    //     return *mpMassMatrix;
+    // }
+
+    SparseMatrixType &GetStiffnessMatrixPrevious()
+    {
+        return *mpStiffnessMatrixPrevious;
+    }
+
+    SparseMatrixType &GetStiffnessMatrix()
+    {
+        return *mpStiffnessMatrix;
+    }
+
+    SparseVectorType &GetRHS()
+    {
+        return *mpRHS;
+    }
+
+    SparseVectorType &GetDx()
+    {
+        return *mpDx;
+    }
+
+    ConvergenceCriteriaType &GetConvergenceCriteria()
+    {
+        return mpConvergenceCriteria;
+    }
+
+    bool GetSolutionFoundFlag()
+    {
+        return mSolutionFound;
+    }
+
+    void SetReformDofSetAtEachStepFlag(bool flag)
+    {
+        this->pBuilderAndSolver()->SetReshapeMatrixFlag(flag);
+    }
+
+    bool GetReformDofSetAtEachStepFlag() const
+    {
+        return this->pBuilderAndSolver()->GetReshapeMatrixFlag();
+    }
+
+    /// Set verbosity level of the solving strategy.
+    /**
+     * - 0 -> mute... no echo at all
+     * - 1 -> print time and basic information
+     * - 2 -> print linear solver data
+     * - 3 -> print debug information
+     */
+    void SetEchoLevel(int Level) override
+    {
+        BaseType::SetEchoLevel(Level);
+        this->pGetEigenSolver()->SetEchoLevel(Level);
+        this->pGetBuilderAndSolver()->SetEchoLevel(Level);
+    }
+
+    /**
+     * Initialization to be performed once before using the strategy.
+     */
+    void Initialize() override
+    {
+        KRATOS_TRY
+
+        ModelPart &rModelPart = BaseType::GetModelPart();
+        const int rank = rModelPart.GetCommunicator().MyPID();
+        typename ConvergenceCriteriaType::Pointer pConvergenceCriteria = mpConvergenceCriteria;
+
+        KRATOS_INFO_IF("PrebucklingStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+            << "Entering Initialize" << std::endl;
+
+        if (mInitializeWasPerformed == false)
+        {
+            SchemePointerType &pScheme = this->pGetScheme();
+
+            if (pScheme->SchemeIsInitialized() == false)
```

Comment:
Idem I prefer ! instead of == false
```suggestion
            if (!pScheme->SchemeIsInitialized())
```

Reply:
Changed